### ISSUE TITLE
Full circle is deformed and flickers

### ISSUE THREAD

**Issue by renyhp** [arp242/find-cursor#22]
I am on Xubuntu 20.04.2 and the suggested command for drawing a full circle on top of the cursor flickers a lot, but also its shape is deformed.

Screenshot from `find-cursor --distance 1 --wait 1000 --line-width 16 --size 100` to see the deformed shape:

![immagine](https://user-images.githubusercontent.com/20142663/108408475-5d52dc80-7225-11eb-91a6-a416d76cdd6d.png)

It looks like every circle for some reason is drawn in two parts: first the semicircle in the diagonal up-left part of the cursor, and then the other one (even when the cursor is not moving); but the second part has smaller radius, and this causes the deformation.

**Comment by arp242** [comment]
If I run it like this:
    
    find-cursor --distance 1 --wait 5000 --line-width 32 --size 100

Then I can kind-of see what you mean: the circle getting smaller isn't quite even, but the deformation isn't as marked as your screenshot though; it's pretty subtle.

This seems a bit of an artificial/contrived example; with `--distance 2` it seems normal. In which use case do you run in to this problem?

> the suggested command for drawing a full circle on top of the cursor flickers a lot

This is probably because it keeps redrawing new windows, instead of creating one window and moving it. I attempted to improve that in 241ae5c, but reverted that in 45e937a because there were some issues with it ([details](https://github.com/arp242/find-cursor/issues/17#issuecomment-617109726))

To be honest this entire mode is a bit of an afterthought; I wrote this just to locate the cursor on my fairly small laptop screen (as in the screenshot), and then people reported they wanted to use it like this to work around problems with some remote desktop solutions: not something I ever considered when I wrote this, but I don't mind supporting it so I added some features/flags for it.

But yeah, it's not optimized for this usage; it was intended as a short-lived "draw window, then quit" kind of thing. I might optimize/improve this at some point, when I feel like it, maybe 😅

**Comment by renyhp** [comment]
Yes, this is a contrived example: I just tried to simulate the behaviour of the "full circle" mode, but bigger and in slow motion. That's because that one also gives this deformation, but it's difficult to capture on a screenshot.

With `--distance 2` I now get that circles are in fact drawn in full (not with those two semicircles), but it still looks the same.

Does it maybe have to do with screen resolution? I'm on a 1920x1080 monitor (that's quite common as a resolution though...)

As for the flickering, thank you, I will try to compile it from that commit and see what happens.

**Comment by arp242** [comment]
> Yes, this is a contrived example: I just tried to simulate the behaviour of the "full circle" mode, but bigger and in slow motion. That's because that one also gives this deformation, but it's difficult to capture on a screenshot.

Ah right.

> Does it maybe have to do with screen resolution? I'm on a 1920x1080 monitor (that's quite common as a resolution though...)

I've got a 1920×1080 screen as well, but it's fairly high density (15"), so maybe it's more pronounced on lower-density screens.
